### PR TITLE
fix: Cards now correctly set color mode for their contents GM-308

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

* Switch Cards to use `Background` to ensure components within cards behave correctly.
* Changes hover behavior to only use borderColor and without an outline for future proofing.

### PR Checklist

- [x] Related to designs:
- [x] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-308
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/Card/index.tsx [hunk 1]
```diff
@@ -55,12 +28,26 @@ const shadowVariants = variant({
...
+>(
+  system.css({ border: 1, borderRadius: '2px' }),
+  shadowVariants,
+  system.states({
+    outline: {
+      '&:hover': {
+        outline: '1px solid currentColor',
+      },
+    },
+  })
```
Comment (codecaaron) [inline packages/gamut/src/Card/index.tsx:46]: @songstephen  We should figure out from a design side how we want to do this stuff going forward - these don't work outside of lightmode currently but will need to.

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://610c44727de0160848db22a4--gamut-preview.netlify.app>https://610c44727de0160848db22a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/610c44727de0160848db22a4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@5.8.1-alpha.d9c4c8.0<br/>@codecademy/gamut-illustrations@0.17.15-alpha.d9c4c8.0<br/>@codecademy/gamut-kit@0.5.17-alpha.d9c4c8.0<br/>@codecademy/gamut-labs@19.19.3-alpha.d9c4c8.0<br/>@codecademy/gamut-patterns@0.5.3-alpha.d9c4c8.0<br/>@codecademy/gamut-styles@11.7.1-alpha.d9c4c8.0<br/>@codecademy/gamut-tests@2.4.18-alpha.d9c4c8.0<br/>@codecademy/gamut@35.12.1-alpha.d9c4c8.0<br/>@codecademy/konamimojisplosion@0.2.49-alpha.d9c4c8.0<br/>@codecademy/markdown-overrides@0.5.61-alpha.d9c4c8.0<br/>@codecademy/styleguide@42.7.1-alpha.d9c4c8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://610c467155581607a499a67e--gamut-preview.netlify.app>https://610c467155581607a499a67e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/610c467155581607a499a67e>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]

**Pull request merged**